docs/developer/designs/topology-awareness/README.md

```diff
@@ -0,0 +1,312 @@
+# Topology Aware Scheduling
+
+## Summary
+
+This design document outlines an approach for implementing topology aware scheduling in KAI-Scheduler, leveraging the topology CRD concepts from Kueue. Topology awareness enables more efficient scheduling of workloads by considering the physical placement of resources within a cluster, leading to improved performance for applications that benefit from resource locality.
```

Comment:
We should provide a reference to kueue's design, either here or in the "Goals" section